Removes unused values from helm chart

ashtongraves opened the pull request.

ashtongraves committed:
Sets default values for GitEndpoint RepoNeedsKey
Used in every hosted ce, discussed with Brian L
https://opensciencegrid.atlassian.net/browse/OPS-363

ashtongraves committed:
Removes dashboard
Discussed with Brian L. No longer used.
https://opensciencegrid.atlassian.net/browse/OPS-363

ashtongraves committed:
Removes HostKeySecret and HostCertSecret
No longer used
https://opensciencegrid.atlassian.net/browse/OPS-363

ashtongraves commented:
@brianhlin @matyasselmeci can one of you give this a look over? Particularly look at the slate section, I wasn't confident if there was still bits being used, I just kind of ripped out everything that referenced slate. Please give it a look over and let me know if there's things we need to keep but rename instead.

ashtongraves commented:
Also is that _helpers.tpl file still used?

brianhlin commented on supported/osg-htc/osg-hosted-ce/Chart.yaml:
```diff
@@ -3,4 +3,4 @@ apiVersion: v1
 description: OSG Hosted Compute Entrypoint
 name: osg-hosted-ce
-version: 4.6.3
+version: 4.6.4
```
There's enough stuff getting ripped out that we should probably bump the minor version. We roughly try to follow semver https://semver.org/#summary

ashtongraves replied:
Bumped to 4.7.0

brianhlin commented on supported/osg-htc/osg-hosted-ce/templates/configmap.yaml:
```diff
@@ -132,24 +131,6 @@ data:
-    {{- range $url, $user := $map }}
-    SCITOKENS /^{{ $url | replace "/" "\\/" | replace "." "\\." | replace "-" "\\-" }}{{ if not ( $url | contains "," ) }},{{ end }}/ {{ $user }}
-    {{- end }}
-    {{- end }}
```
We definitely still need this bit. The name of the ConfigMap can probably be changed (and the ref in the deployment updated) but this is what drives user creation and which organizations are allowed to submit jobs to the CE!

ashtongraves replied:
Renamed instead of deleted

brianhlin commented on supported/osg-htc/osg-hosted-ce/templates/deployment.yaml:
```diff
@@ -47,9 +46,6 @@ spec:
           name: osg-hosted-ce-{{ .Values.Instance }}-htcondor-ce-configuration
-      - name: osg-hosted-ce-{{ .Values.Instance }}-slate-scitokens
-        configMap:
-          name: osg-hosted-ce-{{ .Values.Instance }}-slate-scitokens
```
Same comment about needing this

ashtongraves replied:
Renamed instead of deleted

brianhlin commented on supported/osg-htc/osg-hosted-ce/templates/deployment.yaml:
```diff
@@ -175,22 +141,10 @@ spec:
-        - name: osg-hosted-ce-{{ .Values.Instance }}-slate-scitokens
-          mountPath: /etc/condor-ce/mapfiles.d/50-slate-scitokens.conf
-          subPath: 50-slate-scitokens.conf
-        {{ end }}
```
We still need this

ashtongraves replied:
Renamed instead of deleted

brianhlin commented on supported/osg-htc/osg-hosted-ce/values.yaml:
```diff
@@ -224,10 +208,4 @@ Debug:
-  Instance:
-    ID: "untagged"
-### SLATE-END ###
+  Enabled: false
```
Nitpick: missing newline

ashtongraves replied:
Added newline

brianhlin reviewed (changes requested):
One big issue and a few minor nitpicks. I believe we still use `_helpers.tpl`

ashtongraves committed:
Removes Slate value section and slate references
This chart isn't supporting slate, almost all ce are migrated from slate
https://opensciencegrid.atlassian.net/browse/OPS-363

ashtongraves committed:
Bumps chart version to 4.7.0

ashtongraves commented on supported/osg-htc/osg-hosted-ce/templates/configmap.yaml:
```diff
@@ -71,8 +71,8 @@ data:
-    SLATE_HOSTED_CE = True
-    SCHEDD_ATTRS = $(SCHEDD_ATTRS) SLATE_HOSTED_CE
+    HOSTED_CE = True
+    SCHEDD_ATTRS = $(SCHEDD_ATTRS) HOSTED_CE
```
@brianhlin can you look at this part? Not sure if this is kosher, not really sure what that's used for or if it's expecting something called SLATE_HOSTED_CE.

brianhlin replied:
This shows up in the SchedD ClassAd of the CE:

```
$ condor_ce_status -pool collector.opensciencegrid.org:9619 -schedd -const 'SLATE_HOSTED_CE == True' -af name | head
aei-hypatia-ce1.svc.opensciencegrid.org
amnh-ares-ce1.svc.opensciencegrid.org
amnh-hel-ce1.svc.opensciencegrid.org
amnh-mendel-ce1.svc.opensciencegrid.org
asu-sol-ce1.svc.opensciencegrid.org
canary2.osg.chtc.io
cardiff-igwn-ce1.svc.opensciencegrid.org
chtc-spark-ce1.svc.opensciencegrid.org
clarkson-acres-ce2.svc.opensciencegrid.org
clemson-palmetto-ce1.svc.opensciencegrid.org
```

I'm not sure if anything uses this for queries so it's probably safer to add/append `OSG_HOSTED_CE` and eventually transition to it instead of removing `SLATE_HOSTED_CE`

ashtongraves replied:
Fixed in 305f7f36661e340717be0683c2e97b7ce481bd0e

ashtongraves commented:
@brianhlin can you give this another look over? Also don't merge yet, Jeff D had some concerns with the dashboard piece. Want to make sure he's okay with changes before merge

ashtongraves commented on supported/osg-htc/osg-hosted-ce/templates/configmap.yaml:
```diff
@@ -73,7 +73,8 @@ data:
     SLATE_HOSTED_CE = True
     SCHEDD_ATTRS = $(SCHEDD_ATTRS) SLATE_HOSTED_CE
-
+    OSG_HOSTED_CE = True
```
@brianhlin this work?

brianhlin replied:
I would add it above the `SCHEDD_ATTRS` line for sanity's sake. HTCondor config should generally allow refs to uninitiated values but it can make it harder to read for a human. That way you can combine your line 75 + line 77 into a single line

ashtongraves replied:
Updated, look okay?

brianhlin replied:
Yup, LGTM

ashtongraves replied:
Cool, want to give it one more once over? I might get Jeff D to look too and then merge

ashtongraves replied:
Resolved in 305f7f36661e340717be0683c2e97b7ce481bd0e

brianhlin commented:
FWIW, I added the dashboard piece based on what I thought what Miron wanted but he's decided to go in a separate direction wrt the dashboard.

brianhlin commented on supported/osg-htc/osg-hosted-ce/templates/configmap.yaml:
```diff
@@ -136,15 +137,15 @@ data:
     app.kubernetes.io/instance: {{ .Release.Name }}
 data:
-  50-slate-scitokens.conf: |+
+  50-scitokens.conf: |+
```
It's worth looking in the container to make sure there isn't anything hardcoded referencing this file

ashtongraves replied:
So I searched here: https://github.com/search?q=repo%3Ahtcondor%2Fhtcondor-ce%20scitokens.conf&type=code
And here: https://github.com/search?q=repo%3Aopensciencegrid%2Fdocker-compute-entrypoint%20scitokens.conf&type=code

And don't see any reference at all to this. Is there somewhere else I'm supposed to check? Also what's the difference between that conf and this one: https://github.com/htcondor/htcondor-ce/blob/056e76af28d258db82dc895c0b6a82c02e765fa6/config/mapfiles.d/10-scitokens.conf

brianhlin replied:
Looks like we're good as the CE container images look at the entire mapfiles.d dir for mappings to determine which users need to be provisioned: https://github.com/opensciencegrid/docker-compute-entrypoint/blob/master/base/etc/osg/image-config.d/ce-common-startup

The default conf that you link is mostly intended for admins setting up their own local CE and notice that there aren't any uncommented lines. The entire `mapfiles.d` dir is parsed in alphanumeric order and concatenated together to get credential -> user mappings (the file that determines what's included is `/etc/condor-ce/mapfiles.d`). 

You'll see this pattern used a lot for configuration, e.g. `conf.d` dirs, across different Linux applications.

brianhlin reviewed (approved):
There's one thing that you should cross-check on the container side but otherwise LGTM

ashtongraves commented:
> One big issue and a few minor nitpicks. I believe we still use `_helpers.tpl`

What exactly does that helpers.tpl do? Like it seemed like that line I removed wasn't used right?

brianhlin commented:
`_helpers.tpl` contains named templates: https://helm.sh/docs/chart_template_guide/named_templates/. If you're sure they're not used, you could remove them but it certainly doesn't hurt much if they stick around

ashtongraves commented:
> `_helpers.tpl` contains named templates: https://helm.sh/docs/chart_template_guide/named_templates/. If you're sure they're not used, you could remove them but it certainly doesn't hurt much if they stick around

I'm not sure, I just know for the part I removed, I haven't seen anything namespace related defined with slate-vo- prefix. In fact Looking through the helm chart and our helm release, I don't see anything set for a namespace so not sure where .Release.namespace is coming from, but I'm assuming it was removed at some point and this piece wasn't cleaned up?

ashtongraves committed:
Adds ProxyJumpUser

ashtongraves committed:
Bumps hosted-ce helm to 4.9.0

brianhlin reviewed (approved):
LGTM

ashtongraves committed:
Merge remote-tracking branch 'upstream/main'
